hathor/indexes/manager.py

```diff
@@ -179,6 +188,15 @@ def _manually_initialize(self, tx_storage: 'TransactionStorage') -> None:
 
         tx_storage._update_caches(block_count, tx_count, latest_timestamp, first_timestamp)
 
+    def update(self, tx: BaseTransaction) -> None:
+        """ This is the new update method that indexes should use instead of add_tx/del_tx
+        """
+        # XXX: this _should_ be here, but it breaks some tests, for now this is done explicitly in hathor.manager
```

Comment:
Yes, this isn't too trivial because there are theoretically (but I don't think we have it in practice) cases where we want to remove a transaction but there's no metadata or data that indexes could use to infer that the transaction should be removed regardless of its validation/voided state.

I definitely think there are more improvements to the indexes structure that include this, that we could make on a follow up PR.